### ISSUE TITLE
How to validate object graph as an automated test outside of Unity

### ISSUE THREAD

**Issue by mkrzaczek** [modesttree/Zenject#30]
I am creating an external library (that will be later used in Unity) using Extenject. I would like to check if I can successfully resolve bind objects using an automated test. Eg. I have the following:

`container.Bind<TBind>().To<TTo>().AsSingle();`

To test this simple graph I can just call

`container.Resolve<TBind>()`

and see if there is any exception thrown.

However this is getting much more complicated once using constructs like `WhenInjectedInto`. 
Is there any out-of-the-box solution for this? Or has someone wrote their own working solution and would like to share?

I realize, that there is a Validator when using Extenject withing Unity. But I want to validate the graph without Unity.

**Comment by sbergen** [comment]
As far as I can see, you'd need to construct the `DiContainer` with `isValidating = true`, and then call `ResolveRoots()` once your bindings are in place. I've never used Extenject outside of Unity, so I'm not exactly sure how this would work, or if I'm missing some other details.

**Comment by svermeulen** [comment]
Yes, what @sbergen said.  Also, make sure you set the validation error response to 'throw' instead of logging, if executing inside unity tests.  Also note the `RootResolveMethods` field.

For example:
```
var container = new DiContainer(true);
container.Settings = new ZenjectSettings(
    ValidationErrorResponses.Throw, RootResolveMethods.All);

// Add bindings 

// This will throw if validation fails
container.ResolveRoots()
```

The RootResolveMethods will determine whether it only validates the actual object graph (constructed based on the NonLazy root bindings) or whether it will attempt to resolve all bindings regardless of whether they are inside the object graph or not.
